### PR TITLE
Divorce JSON and KCL

### PR DESCRIPTION
**Pull request by adamchalmers**
Removes JSON from the KCL object model. Closes https://github.com/KittyCAD/modeling-app/issues/1130 -- it was filed on Nov 27 last year. Hopefully I close it before its one year anniversary.

Changes:

- Removed the UserVal variant from `enum KclValue`. That variant held JSON data.
- Replaced it with several new variants like Number, String, Array (of KCL values), Object (where keys are String and values are KCL values)
- Added a dedicated Sketch variant to KclValue. We used to have a variant like this, but I removed it as an experimental approach to fix this issue. Eventually I decided to undo it and use the approach of this PR instead.
- Removed the `impl_from_arg_via_json` macro, which implemented conversion from KclValue to Rust types by matching the KclValue to its UserVal variant, grabbing the JSON, then deserializing that into the desired Rust type. 
- Instead, replaced it with manual conversion from KclValue to Rust types, using some convenience macros like `field!`

### CODE REVIEW

File: src/wasm-lib/kcl/src/ast/types/execute.rs [hunk 1]
```diff
@@ -42,25 +45,19 @@ impl Node<MemberExpression> {
             }
         };
 
-        let array_json = array.get_json_value()?;
+        // let array_json = array.get_json_value()?;
```
Comment (jtran) [inline src/wasm-lib/kcl/src/ast/types/execute.rs:48]:
```suggestion
```

File: src/components/ModelingSidebar/ModelingPanes/MemoryPane.tsx [hunk 2]
```diff
@@ -111,7 +111,7 @@ export const processMemory = (programMemory: ProgramMemory) => {
         ?.map?.(({ identifier }: any) => identifier?.name || '')
         .join(', ')})__`
     } else {
-      processedMemory[key] = val.value
```
Comment (adamchalmers) [inline src/components/ModelingSidebar/ModelingPanes/MemoryPane.tsx:114]: @jtran @franknoirot Hey, on this line `val` is now the type `never`. So I'm not sure what to do here, it basically seems to be saying that this final `else` branch will never actually be triggered. Any suggestions for what to do here? Should I just `return processedMemory`?

File: src/wasm-lib/kcl/src/ast/types/execute.rs [hunk 3]
```diff
@@ -759,35 +775,37 @@ impl Property {
...
+        }
+        KclValue::Number{value: num, meta:_} => {
+            let num = *num;
+            if num < 0.0 {
+                return make_err(format!("'{num}' is negative, so you can't index an array with it"))
+            }
+            let nearest_int = num.round();
+            let delta = num-nearest_int;
+            if delta < FLOAT_TO_INT_MAX_DELTA {
+                Ok(Property::Number(nearest_int as usize))
```
[the remark this answers is not in the document]
Comment (adamchalmers) [reply src/wasm-lib/kcl/src/ast/types/execute.rs:803]: In this limited context of indexing an array, yeah. Broader changes for future PRs.

File: src/wasm-lib/kcl/tests/helix_ccw/program_memory.snap
[the remark this answers is not in the document]
Comment (adamchalmers) [reply src/wasm-lib/kcl/tests/helix_ccw/program_memory.snap:1]: *some* of the duplicates :)

File: src/wasm-lib/tests/executor/inputs/member_expression_sketch.kcl [hunk 4]
```diff
@@ -13,5 +13,5 @@ sq = square([0,0], 10)
 cb = square([3,3], 4)
     |> extrude(10, %)
 
-pt1 = sq.paths[0]
-pt2 = cb.value[0]
+// pt1 = sq.paths[0]
+// pt2 = cb.value[0]
```
[the remark this answers is not in the document]
Comment (adamchalmers) [reply src/wasm-lib/tests/executor/inputs/member_expression_sketch.kcl:17]: Yeah. I never intended that people would be able to access these properties. I wrote this in the test just to run a test, but now the simulation test framework lets us do this in a better way.

File: src/components/ModelingSidebar/ModelingPanes/MemoryPane.tsx [hunk 5]
```diff
@@ -89,7 +89,7 @@ export const processMemory = (programMemory: ProgramMemory) => {
   const processedMemory: any = {}
   for (const [key, val] of programMemory?.visibleEntries()) {
     if (
-      (val.type === 'UserVal' && val.value.type === 'Sketch') ||
+      val.type === 'Sketch' ||
       // @ts-ignore
       (val.type !== 'Function' && val.type !== 'UserVal')
```
Comment (jtran) [inline src/components/ModelingSidebar/ModelingPanes/MemoryPane.tsx:94]: All these conditions can be cleaned up, but the main thing is that there shouldn't be mention of `UserVal` anymore.

```suggestion
      val.type !== 'Function'
```

**Review by jtran — APPROVED** [review]
Might've broken some error messages in that last commit. But once that's resolved, I think it looks good!

**Pull request merged**